crates/stackable-operator/src/commons/authentication/oidc.rs

```diff
@@ -132,10 +138,38 @@ impl AuthenticationProvider {
             })?;
         }
 
-        url.set_path(&self.root_path);
         Ok(url)
     }
 
+    /// Returns the OIDC endpoint [`Url`] without a trailing slash.
+    ///
+    /// To get the well-known URL, please use [`Self::well_known_url`].
+    pub fn endpoint_url(&self) -> Result<Url> {
+        let mut url = self.base_url()?;
+        // Some tools can not cope with a trailing slash, so let's remove that
+        url.set_path(self.root_path.trim_end_matches('/'));
+        Ok(url)
+    }
+
+    /// Returns the well-known [`Url`] without a trailing slash.
+    ///
+    /// It is basically the [`Self::endpoint_url`] joined with
+    /// "./.well-known/openid-configuration", while watching out for URL joining madness.
+    pub fn well_known_url(&self) -> Result<Url> {
+        let mut url = self.base_url()?;
+
+        // Url::join cuts of the part after the last slash :/
+        // So we need to make sure we have a trailing slash, so that nothing get's cut of.
+        let mut root_path_with_trailing_slash = self.root_path.trim_end_matches('/').to_string();
+        root_path_with_trailing_slash.push('/');
+        url.set_path(&root_path_with_trailing_slash);
+        url.join(DEFAULT_OIDC_WELLKNOWN_PATH)
+            .with_context(|_| JoinPathSnafu {
+                url: url.clone(),
+                path: DEFAULT_OIDC_WELLKNOWN_PATH.to_owned(),
+            })
+    }
```

Comment:
suggestion: This can be slightly improved.

```suggestion
        /// Returns the well-known OIDC configuration [`Url`] without a trailing slash.
    ///
    /// The returned url is a combination of [`Self::endpoint_url`] joined with
    /// the well-known OIDC configuration path [`DEFAULT_WELLKNOWN_OIDC_CONFIG_PATH`].
    pub fn well_known_config_url(&self) -> Result<Url> {
        let mut url = self.base_url()?;

        // Taken from https://docs.rs/url/latest/url/struct.Url.html#method.join:
        // A trailing slash is significant. Without it, the last path component is considered to be
        // a “file” name to be removed to get at the “directory” that is used as the base.
        //
        // Because of that behaviour, we first need to make sure that the root path doesn't contain
        // any trailing slashes to finally append the well-known config path to the url. The path
        // already contains a prefixed slash.
        let mut root_path_with_trailing_slash = self.root_path.trim_end_matches('/').to_string();
        root_path_with_trailing_slash.push_str(DEFAULT_WELLKNOWN_OIDC_CONFIG_PATH);
        url.set_path(&root_path_with_trailing_slash);

        Ok(url)
    }
```

Reply:
Hmm, although I don't particularly like the Url::join function, I think trying to use an library is preferred over concatenating the string ourselves.
The suggestion can e.g. produce invalid URLs when `root_path` is not URL escaped

Reply:
PR updated to proposal in 9cea263